os/pm/pm_procfs.c

```diff
@@ -357,45 +335,40 @@ static size_t power_metrics_read(FAR struct file *filep, FAR char *buffer, size_
 	struct pm_time_in_each_s mtrics;
 	size_t copysize;
 	size_t totalsize;
-	int domain;
-	int index;
 
 	priv = (FAR struct power_file_s *)filep->f_priv;
 
-	index = 0;
 	copysize = 0;
 	totalsize = 0;
 
 	if (priv->offset == 0) {
-		domain = priv->dir.domain;
 
-		if (domain >= 0 && domain < CONFIG_PM_NDOMAINS) {
-			pm_get_domainmetrics(domain, &mtrics);
+		pm_get_domainmetrics(&mtrics);
 
-			/* Time in NORMAL state */
-			copysize = snprintf(buffer, buflen, " Time in %s : %d\n", g_power_states[index++], mtrics.normal);
-			buflen -= copysize;
-			buffer += copysize;
-			totalsize += copysize;
+		/* Time in NORMAL state */
+		copysize = snprintf(buffer, buflen, " Time in %s : %d\n", g_power_states[index++], mtrics.normal);
```

Comment:
In here, you removed the variable declaration above (ie. index). But it was still being used. Did you enable CONFIG_PM_METRICS when you modified this?

Reply:
On the other hand, the code piece seems abit redundant. Since there is only 4 PM states.
```
for (int index= 0; index < CONFIG_PM_STATES; index++) {
                switch (index) {
                // Each case get different mtrics state
                }
                copysize = snprintf(buffer, buflen, " Time in %s : %d\n", g_power_states[index], // Set final argument from result above);
		buflen -= copysize;
		buffer += copysize;
		totalsize += copysize;
}

```
CONFIG_PM_STATES is just a definition to prevent hardcoding, it might not be defined yet

Reply:
Done

Reply:
This seems to be reverted, could you please check again?